core/src/consensus.rs

```diff
@@ -648,6 +795,121 @@ impl Tower {
             None
         }
     }
+
+    pub fn get_filename(path: &Path, node_pubkey: &Pubkey) -> PathBuf {
+        PathBuf::from(path)
+            .join(format!("tower-{}", node_pubkey))
+            .with_extension("bin")
+    }
+
+    pub fn save(&self, node_keypair: &Arc<Keypair>) -> Result<()> {
+        if self.node_pubkey != node_keypair.pubkey() {
+            return Err(TowerError::WrongTower(format!(
+                "node_pubkey is {:?} but found tower for {:?}",
+                node_keypair.pubkey(),
+                self.node_pubkey
+            )));
+        }
+
+        fs::create_dir_all(&self.save_path)?;
+        let filename = Self::get_filename(&self.save_path, &self.node_pubkey);
+        let new_filename = filename.with_extension("new");
+        {
+            let mut file = File::create(&new_filename)?;
+            let saveable_tower = SavedTower::new(self, node_keypair)?;
+            bincode::serialize_into(&mut file, &saveable_tower)?;
+            //file.sync_all().unwrap();
```

Comment:
Well, it turned out this and the other `.sync_all()` causes a lot of instability even with local cluster tests...